Common/Securities/BuyingPowerModel.cs

```diff
@@ -267,38 +270,38 @@ public virtual HasSufficientBuyingPowerForOrderResult HasSufficientBuyingPowerFo
 
                     // we continue with this call for underlying
                     return underlying.BuyingPowerModel.HasSufficientBuyingPowerForOrder(
-                        new HasSufficientBuyingPowerForOrderParameters(parameters.Portfolio, underlying, newOrder));
+                        parameters.ForUnderlying(newOrder)
+                    );
                 }
 
-                return new HasSufficientBuyingPowerForOrderResult(true);
+                return parameters.Sufficient();
             }
 
             // When order only reduces or closes a security position, capital is always sufficient
             if (parameters.Security.Holdings.Quantity * parameters.Order.Quantity < 0 && Math.Abs(parameters.Security.Holdings.Quantity) >= Math.Abs(parameters.Order.Quantity))
             {
-                return new HasSufficientBuyingPowerForOrderResult(true);
+                return parameters.Sufficient();
             }
 
             var freeMargin = GetMarginRemaining(parameters.Portfolio, parameters.Security, parameters.Order.Direction);
             var initialMarginRequiredForOrder = GetInitialMarginRequiredForOrder(
-                new InitialMarginRequiredForOrderParameters(parameters.Portfolio.CashBook,
-                    parameters.Security,
-                    parameters.Order));
+                new InitialMarginRequiredForOrderParameters(
+                    parameters.Portfolio.CashBook, parameters.Security, parameters.Order
```

Comment:
I'd like to go through all of these model implementations and remove all the noise from the parameter and result classes. One pattern already used on some is result methods, such as this `Sufficient()` method directly on the parameter type. Another pattern to clean up calls to other model functions is implicit conversions. In this case, it would be a implicit conversion defined in `HasSufficientBuyingPowerForOrderParameters` that converts to `InitialMarginRequiredForOrderParameters` resulting in noise reduction:
``` csharp
var initialMarginRequiredForOrder = GetInitialMarginRequiredForOrder(parameters);
```